src/layout.rs

```diff
@@ -591,6 +608,200 @@ fn try_split(area: Rect, layout: &Layout) -> Result<Rc<[Rect]>, AddConstraintErr
+        assert_eq!(chunks[0].height, 3);
+        assert_eq!(chunks[1].height, 4);
+        assert_eq!(chunks[2].height, 3);
+    }
+
+    #[test]
+    fn test_split_equally_in_overconstrained_case_for_min() {
+        let target = Rect {
+            x: 2,
+            y: 2,
+            width: 100,
+            height: 10,
+        };
+
+        let chunks = Layout::default()
+            .direction(Direction::Horizontal)
+            .constraints(
+                [
+                    Constraint::Percentage(50),
+                    Constraint::Min(10),
+                    Constraint::Percentage(50),
+                ]
+                .as_ref(),
+            )
+            .split(target);
+
+        assert_eq!(target.width, chunks.iter().map(|r| r.width).sum::<u16>());
+        assert_eq!(chunks[0].width, 50);
+        assert_eq!(chunks[1].width, 10);
+        assert_eq!(chunks[2].width, 40);
+
+        let chunks = Layout::default()
+            .direction(Direction::Horizontal)
+            .constraints(
+                [
+                    Constraint::Percentage(50),
+                    Constraint::Min(10),
+                    Constraint::Percentage(50),
+                ]
+                .as_ref(),
+            )
+            .prefer_equal_chunks(true)
+            .split(target);
+
+        assert_eq!(target.width, chunks.iter().map(|r| r.width).sum::<u16>());
+        assert_eq!(chunks[0].width, 45);
+        assert_eq!(chunks[1].width, 10);
+        assert_eq!(chunks[2].width, 45);
+    }
+
+    #[test]
+    fn test_split_equally_in_overconstrained_case_for_max() {
+        let target = Rect {
+            x: 2,
+            y: 2,
+            width: 100,
+            height: 10,
+        };
+
+        let chunks = Layout::default()
+            .direction(Direction::Horizontal)
+            .constraints(
+                [
+                    Constraint::Percentage(30),
+                    Constraint::Max(10),
+                    Constraint::Percentage(30),
+                ]
+                .as_ref(),
+            )
+            .split(target);
+
+        assert_eq!(target.width, chunks.iter().map(|r| r.width).sum::<u16>());
+        assert_eq!(chunks[0].width, 30);
+        assert_eq!(chunks[1].width, 10);
+        assert_eq!(chunks[2].width, 60);
+
+        let chunks = Layout::default()
+            .direction(Direction::Horizontal)
+            .constraints(
+                [
+                    Constraint::Percentage(30),
+                    Constraint::Max(10),
+                    Constraint::Percentage(30),
+                ]
+                .as_ref(),
+            )
+            .prefer_equal_chunks(true)
+            .split(target);
+
+        assert_eq!(target.width, chunks.iter().map(|r| r.width).sum::<u16>());
+        assert_eq!(chunks[0].width, 33);
+        assert_eq!(chunks[1].width, 34);
+        assert_eq!(chunks[2].width, 33);
+    }
```

Comment:
I don't understand why this test is behaving this way. This seems like a bug to me.

Reply:
Fixed in https://github.com/ratatui-org/ratatui/pull/395/commits/0b41dac3944df4c052ab872385d6f2bf926d3dd1

This commit changes the implementation of `Max` and `Min` by adding a `MEDIUM` constraint instead of a `WEAK` constraint. It should result in the same layouts for us though, because there were no other `MEDIUM` constraints used.

Reply:
Good fix!